### PR TITLE
[RFR] Allow real time update of resources

### PR DESCRIPTION
**Pull request by djhi**
This requires https://github.com/marmelab/admin-on-rest/pull/318
- [x] Implement a saga which starts observable apollo queries on graphql endpoint
- [x] Refactor the code and clean it
- [x] Tests

screen

### CODE REVIEW

File: src/realtime/saga.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        }
+    };
+
+export const watchLocationChangeFactory = watchCrudActions => function* watchLocationChange(action) {
+    yield takeLatest([CRUD_GET_LIST, CRUD_GET_ONE], watchCrudActions);
+};
+
+export default (apolloConfiguredClient, apolloWatchParameters) => function* aorGraphQlSaga() {
+    const watchCrudActions = watchCrudActionsFactory(apolloConfiguredClient, apolloWatchParameters);
+    yield takeLatest(LOCATION_CHANGE, watchLocationChangeFactory(watchCrudActions));
```
Comment (djhi) [inline src/realtime/saga.js]: I'm not even sure this is needed. I might just listen to `CRUD_GET_LIST` and `CRUD_GET_ONE`

File: src/buildApolloConfiguredClient.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+export const buildQueryObserver = (type, resource, apolloParams, observer) => ({
+    complete() {
+        observer.complete();
+    },
+    error() {
+        observer.error();
+    },
+    next(apolloQueryResult) {
+        console.log('QueryObserver.next', { apolloQueryResult });
```
Comment (fzaninotto) [inline src/buildApolloConfiguredClient.js]: `console.log()` spotted

Comment (djhi) [inline src/buildApolloConfiguredClient.js]: See point 2 in todo list ^^

File: package.json [hunk 3]
```diff
@@ -14,6 +14,7 @@
     "watch-test": "make watch-test"
   },
   "devDependencies": {
+    "admin-on-rest": "^0.8.1",
```
Comment (fzaninotto) [inline package.json]: use tilde instead

File: package.json [hunk 4]
```diff
@@ -33,13 +34,15 @@
...
+    "lodash.pick": "4.4.0",
     "lodash.pickby": "4.6.0",
     "pluralize": "3.1.0",
     "react-router": "3.0.2",
     "react-router-redux": "4.0.8",
     "redux-saga": "0.14.3"
   },
   "peerDependencies": {
-    "admin-on-rest": "~0.8.0"
+    "admin-on-rest": "^0.8.1"
```
Comment (fzaninotto) [inline package.json]: already in `devDependencies`?

Comment (djhi) [inline package.json]: It serves different purpose:

`peerDependencies` indicates to the end user that this package is depending on aor 
However, npm and yarn do not install peer deps in development mode

`devDependencies` indicates to the developer contributing to this package that it needs aor to run tests for example

**Pull request merged**